{"priority": "minor", "kind": "bug", "repository": {"links": {"self": {"href": "data/repositories/osrf/srcsim.json"}, "html": {"href": "#!/osrf/srcsim"}, "avatar": {"href": "data/bytebucket.org/ravatar/{c41021d9-0217-4628-999d-7f8d82f98dcb}ts=c_plus_plus"}}, "type": "repository", "name": "srcsim", "full_name": "osrf/srcsim", "uuid": "{c41021d9-0217-4628-999d-7f8d82f98dcb}"}, "links": {"attachments": {"href": "data/repositories/osrf/srcsim/issues/23/attachments_page=1.json"}, "self": {"href": "data/repositories/osrf/srcsim/issues/23.json"}, "watch": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/srcsim/issues/23/watch"}, "comments": {"href": "data/repositories/osrf/srcsim/issues/23/comments_page=1.json"}, "html": {"href": "#!/osrf/srcsim/issues/23/playback-error"}, "vote": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/srcsim/issues/23/vote"}}, "reporter": {"display_name": "Tom Tseemceeb Xiong", "uuid": "{07f3561e-974e-4e4d-962c-56f955baaac3}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B07f3561e-974e-4e4d-962c-56f955baaac3%7D"}, "html": {"href": "https://bitbucket.org/%7B07f3561e-974e-4e4d-962c-56f955baaac3%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/6aaf7437b4e057d91bcfca8bb6b456c9d=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsTX-1.png"}}, "nickname": "Tom_Xiong", "type": "user", "account_id": "557058:5667b998-d71d-4e95-94f3-a06536d031b0"}, "title": "Playback error", "component": null, "votes": 0, "watches": 1, "content": {"raw": "When I run the command: \r\n```\r\n#!\r\ngazebo -p ~/.gazebo/log/<timestamp>/gzserver/state.log\r\n```\r\nI get the error:\r\n```\r\n#!\r\n[FATAL] [1479541568.200997765]: A ROS node for Gazebo has not been initialized, unable to load plugin. \r\nLoad the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in the gazebo_ros package)\r\n```\r\n\r\nI'm unsure how to fix this issue. I only have this error when I attempt to play back valkyrie ie. (qual1, qual2) logs. If I simply just log in gazebo of a random object I don't receive this issue. But otherwise I can't review my srcsim log files.\r\n\r\nFix: So apparently to run your log files, you run the command: \r\n\r\n```\r\n#!\r\n\r\nroscore\r\n```\r\nThen in a new terminal run the command:\r\n\r\n```\r\n#!\r\nrosrun gazebo_ros gazebo -p ~/.gazebo/log/<timestamp>/gzserver/state.log\r\n\r\n```\r\nThis will play back your log files without the error. Suggestion to update wiki. \r\nI also encountered an error after playing your log file. For some reason the gzserver client doesn't close after quitting the gui. So I had to manually do it through system monitor. \r\n", "markup": "markdown", "html": "<p>When I run the command: </p>\n<div class=\"codehilite\"><pre><span></span>gazebo -p ~/.gazebo/log/&lt;timestamp&gt;/gzserver/state.log\n</pre></div>\n\n\n<p>I get the error:</p>\n<div class=\"codehilite\"><pre><span></span>[FATAL] [1479541568.200997765]: A ROS node for Gazebo has not been initialized, unable to load plugin. \nLoad the Gazebo system plugin &#39;libgazebo_ros_api_plugin.so&#39; in the gazebo_ros package)\n</pre></div>\n\n\n<p>I'm unsure how to fix this issue. I only have this error when I attempt to play back valkyrie ie. (qual1, qual2) logs. If I simply just log in gazebo of a random object I don't receive this issue. But otherwise I can't review my srcsim log files.</p>\n<p>Fix: So apparently to run your log files, you run the command: </p>\n<div class=\"codehilite\"><pre><span></span>roscore\n</pre></div>\n\n\n<p>Then in a new terminal run the command:</p>\n<div class=\"codehilite\"><pre><span></span>rosrun gazebo_ros gazebo -p ~/.gazebo/log/&lt;timestamp&gt;/gzserver/state.log\n</pre></div>\n\n\n<p>This will play back your log files without the error. Suggestion to update wiki. \nI also encountered an error after playing your log file. For some reason the gzserver client doesn't close after quitting the gui. So I had to manually do it through system monitor. </p>", "type": "rendered"}, "assignee": null, "state": "resolved", "version": null, "edited_on": null, "created_on": "2016-11-19T07:52:57.503798+00:00", "milestone": null, "updated_on": "2016-11-20T19:27:52.274768+00:00", "type": "issue", "id": 23}